File: src/main/java/org/team1/nbe1_3_team01/domain/board/constants/CommonBoardType.java
package org.team1.nbe1_3_team01.domain.board.constants;

import lombok.Getter;
import org.team1.nbe1_3_team01.domain.board.exception.NotFoundTypeException;

import static org.team1.nbe1_3_team01.global.util.ErrorCode.CATEGORY_NOT_FOUND;

@Getter
public enum CommonBoardType {

    NOTICE("y"),
    STUDY("n");

    private final String type;

    CommonBoardType(String type) {
        this.type = type;
    }

    public static CommonBoardType getType(String type) {
        for (CommonBoardType enumType : values()) {
            if(enumType.getType().equals(type)) {
                return enumType;
            }
        }
        throw new NotFoundTypeException(CATEGORY_NOT_FOUND);
    }
}

File: src/main/java/org/team1/nbe1_3_team01/domain/board/constants/CommonBoardType.kt
package org.team1.nbe1_3_team01.domain.board.constants

import lombok.Getter
import org.team1.nbe1_3_team01.domain.board.exception.NotFoundTypeException
import org.team1.nbe1_3_team01.global.util.ErrorCode

@Getter
enum class CommonBoardType(private val type: String) {
    NOTICE("y"),
    STUDY("n");

    companion object {
        fun getType(type: String): CommonBoardType {
            for (enumType in entries) {
                if (enumType.type == type) {
                    return enumType
                }
            }
            throw NotFoundTypeException(ErrorCode.CATEGORY_NOT_FOUND)
        }
    }
}
